src/app/view-models/social-media-post-extended.ts

```diff
@@ -14,24 +14,34 @@ export class SocialMediaPostExtended implements SocialMediaPost {
         this.sortOrder = post.sortOrder;
         this.timestamp = post.timestamp;
         this.mediaType = this.getMediaType(post.url);
+        this.postType = this.getPostType(post.url);
     }
 
-    getMediaType(url:string): string{
+    getMediaType(url: string): string {
         let type: string;
         type = '';
-        if (url.indexOf('facebook')>=0)
-        {
+        if (url.indexOf('facebook') >= 0) {
             type = 'Facebook';
-        }
-        else if (url.indexOf('twitter')>=0)
-        {
+        } else if (url.indexOf('twitter') >= 0 ) {
             type = 'Twitter';
-        }
-        else if (url.indexOf('instagram') >=0)
-        {
+        } else if (url.indexOf('instagram') >= 0) {
             type = 'Instagram';
         }
         return type;
     }
 
-}
\ No newline at end of file
+    // for facebook only, the html tag is different for post and video
```

Comment:
maybe we should rename the function to getFacebookPostType and do the if url contains facebook check in the constructor before this function is called